### ISSUE TITLE
DRP fails on file names that do not follow the standard pattern

### ISSUE THREAD

**Issue by MosfireDRP** [Mosfire-DataReductionPipeline/MosfireDRP#9]
Alan Stockton reported today that the pipeline was failing on combining the wavelength files.
The error was:

Traceback (most recent call last):
  File "driver.py", line 23, in <module>
    Wavelength.imcombine(obsfiles, maskname, band, waveops)
  File "/Volumes/T1/mosdrp/MOSFIRE/Wavelength.py", line 208, in imcombine
    wavename = filelist_to_wavename(files, bandname, maskname, options)
  File "/Volumes/T1/mosdrp/MOSFIRE/Wavelength.py", line 80, in filelist_to_wavename
    end = files[-1].split('/')[-1].rstrip(".fits").split("_")[1]
IndexError: list index out of range

After trying a few different things, I asked him to send me the Offset files, and I found that he was using the standard KOA names, rather than the mosfire names.

Example:

MF.20120911.50084.fits

instead of m20140911_00084.fits

I asked him to use the KOA script to rename the files and after that the pipeline worked.

A note should be added to the manual to this effect.
Also, we might want to consider making the pipeline less sensitive to specific file names, by using information contained in the headers (UT date and such).

**Comment by anstockton** [comment]
Hi Luca,

One last (I hope) question:

I’m at the final rectify step, but I am unsure exactly what the 2nd quoted file in the following is supposed to be:

Rectify.handle_rectification(maskname, redfiles,
        'lambda_solution_wave_stack_J_m130114_0443-0445.fits',
    band,    "/Users/mkassis/Documents/KeckInstrs/MOSFIRE/DRP_CODE_June2014/DATA/2014may08/m130114_0443.fits",
    waveops)

It appears to be a single file, but, at least to me, the manual is not very clear on which file it is.

Thanks again,  

Alan Stockton
Institute for Astronomy
University of Hawaii

**Comment by lucarizzi** [comment]
Alan

You can chose any of the science files. 
The specific file you chose does not affect the reduction. 

It is used to inform the pipeline of the list of objects in your slits by providing access to the so called CSU extension of the fits file 

Luca

**Comment by anstockton** [comment]
OK, thanks.

Alan  

~~~~~~~~~~Alan Stockton
Institute for Astronomy
University of Hawaii


On Thursday, January 15, 2015 at 6:37 PM, lucarizzi wrote:

> Alan  
>  
> You can chose any of the science files.  
> The specific file you chose does not affect the reduction.  
>  
> It is used to inform the pipeline of the list of objects in your slits by providing access to the so called CSU extension of the fits file  
>  
> Luca  
>  
>  
>  
> > On Jan 15, 2015, at 3:41 PM, anstockton <notifications@github.com (mailto:notifications@github.com)> wrote:  
> >  
> > Hi Luca,  
> >  
> > One last (I hope) question:  
> >  
> > I’m at the final rectify step, but I am unsure exactly what the 2nd quoted file in the following is supposed to be:  
> >  
> > Rectify.handle_rectification(maskname, redfiles,  
> > 'lambda_solution_wave_stack_J_m130114_0443-0445.fits',  
> > band, "/Users/mkassis/Documents/KeckInstrs/MOSFIRE/DRP_CODE_June2014/DATA/2014may08/m130114_0443.fits",  
> > waveops)  
> >  
> > It appears to be a single file, but, at least to me, the manual is not very clear on which file it is.  
> >  
> > Thanks again,  
> >  
> > Alan Stockton  
> > Institute for Astronomy  
> > University of Hawaii  
> >  
> >  
> > On Wednesday, January 14, 2015 at 6:02 PM, MosfireDRP wrote:  
> >  
> > > Closed #9 (https://github.com/Mosfire-DataReductionPipeline/MosfireDRP/issues/9).  
> > >  
> > > —  
> > > Reply to this email directly or view it on GitHub (https://github.com/Mosfire-DataReductionPipeline/MosfireDRP/issues/9#event-219203251).  
> > >  
> > >  
> > >  
> > >  
> > >  
> > —  
> > Reply to this email directly or view it on GitHub.  
> >  
>  
> —
> Reply to this email directly or view it on GitHub (https://github.com/Mosfire-DataReductionPipeline/MosfireDRP/issues/9#issuecomment-70207383).
>  
>  
>  
>  
>
